### PR TITLE
Add test from PR 50799 for impl PR 56853

### PR DESCRIPTION
**Pull request by bernd5**
With PR #56853 we got support for new lines in string interpolations holes. That PR has some semantic and error tests (for C# < 11) but no syntax test. Therefore I would like to add the test from the closed PR https://github.com/dotnet/roslyn/pull/50799.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs [hunk 1]
```diff
@@ -207,6 +207,33 @@ void M()
             EOF();
         }
 
+        [Fact]
+        public void TestInterpolatedStringWithNewLinesInExpression()
+        {
+            var text = @"$""Text with {\nnew[] {\n 1, 2, 3 \n}[2]\n}parts and new line expressions!""";
+            text = text.Replace(@"\n", System.Environment.NewLine);
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs:214]: I don't see the need for this replacement call

File: src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs [hunk 2]
```diff
@@ -207,6 +207,33 @@ void M()
...
+
+            var expr = this.ParseExpression(text);
+
+            if (expr is InterpolatedStringExpressionSyntax intStr)
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs:218]: Just did a direct cast

File: src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs (hunk 1 above)
Comment (bernd5) [reply src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs]: it is a @" string - without escapes...

Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs]: gah  even more confusing.  I hate it :)

File: src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs [hunk 3]
```diff
@@ -207,6 +207,78 @@ void M()
             EOF();
         }
 
+        [Fact]
+        public void TestInterpolatedStringWithNewLinesInExpression()
+        {
+            var text = @"$""Text with {\nnew[] {\n 1, 2, 3 \n}[2]\n}parts and new line expressions!"""
+                .Replace("\\n", System.Environment.NewLine);
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs:214]: I don't understand the need for this. Why are we not just putting the newlines in the verbatim string?

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
@CyrusNajmabadi @dotnet/roslyn-compiler for another review.

**Comment by bernd5** [comment]
@CyrusNajmabadi could you have another look?

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @bernd5! Sorry it took a bit to get this merged, we had some infra troubles the past week.

**Pull request merged**